server/resp.go

```diff
@@ -197,6 +200,25 @@ func (c *Conn) writeFieldValues(fv []FieldValue) error {
 	return c.WritePacket(data)
 }
 
+func (c *Conn) writeBinlogEvents(s *replication.BinlogStreamer) error {
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		ev, err := s.GetEvent(ctx)
+		cancel()
+
+		if err == context.DeadlineExceeded {
+			continue
+		}
+		data := make([]byte, 4, 32)
+		data = append(data, OK_HEADER)
+
+		data = append(data, ev.RawData...)
```

Comment:
and please add some reference to binlog protocal, such as https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_replication.html